lib/dictBuilder/zdict.c

```diff
@@ -969,15 +969,16 @@ static size_t ZDICT_addEntropyTablesFromBuffer_advanced(
 }
 
 /* Hidden declaration for dbio.c */
-size_t ZDICT_trainFromBuffer_unsafe_legacy(
+ZDICTLIB_API size_t ZDICT_trainFromBuffer_unsafe_legacy(
```

Comment:
I'm surprised, normally it's enough for `ZDICTLIB_API` to be on the declaration side (`zdict.h`). It shouldn't be needed on the implementation side (`zdict.c`).